cuckoo/processing/memory.py

```diff
@@ -1020,12 +1025,14 @@ def run(self):
         @return: volatility results dict.
         """
         self.key = "memory"
+        task_machine = self.task["guest"]["name"]
+        machine_manager = self.task["guest"]["manager"].lower()
 
         results = {}
         if HAVE_VOLATILITY:
             if self.memory_path and os.path.exists(self.memory_path):
                 try:
-                    results = VolatilityManager(self.memory_path).run()
+                    results = VolatilityManager(self.memory_path).run(manager=machine_manager, vm=task_machine)
```

Comment:
I'd propose to do the profile selection in this class already and pass it along properly to `VolatilityManager`, which will do its own thing (namely the whole interaction with Volatility). Don't forget to include some basic unit tests for the `profile` selection process.